onnx/shape_inference/implementation.cc

```diff
@@ -140,6 +150,10 @@ void mergeShapesAndTypes(const TypeProto_Tensor& inferredType, TypeProto_Tensor*
     auto* existingDim = existingType->mutable_shape()->mutable_dim(i);
     if (!existingDim->has_dim_value()) {
       *existingDim = inferredDim;
+      // If shape cannot be inferred; set it as a symbolic shape
+      if (inferredType.shape().dim(i).dim_value() == 0 && !inferredType.shape().dim(i).has_dim_param()) {
+        existingDim->set_dim_param(SymbolicShape::createNew());
```

Comment:
how do you createNew simply generated a random sumbol right... how do you guarantee this symbol is unique within the graph? You need to maintain a graph level symbol list and make sure every new symbols is indeed unique for this graph